### DISCUSSION TITLE
How to crop images

### DISCUSSION THREAD

**Discussion by macCesar** [dietercoopman/smart#17] [Q&A] [answered]
Is there a way to auto-crop images by setting width and height ??

`<x-smart-image width="200" height="400"  src="original-1920x1080-image-file" />`

Maybe you can use the `fit` command to crop and resize at the same time and use a `data-position` attribute to set a position where cutout will be positioned, like:

- top-left
- top
- top-right
- left
- center (default)
- right
- bottom-left
- bottom
- bottom-right

**BTW, Thanks for this excellent component!!!**

**Comment by dietercoopman** [comment, marked as answer]
Hi @macCesar you can crop images with a template , this is an example using this configuration 

```php
return [
    'image' => [
        'path'      => 'smart',
        'templates' => [
            'crop' => [
                'resize'    => [null, 200, ['aspectRatio']],
                'crop' => [100, 100, 25, 25]
            ]
        ]
    ]
];
```
This is the result
![Schermafbeelding 2021-12-16 om 07 14 09](https://user-images.githubusercontent.com/4672752/146318299-f2e99b9a-25dc-47f2-acf2-500db2583d30.png)

Kind regards Dieter

**Reply by macCesar** [reply]
Excellent... Thanks 👍🏻
